visidata/aggregators.py

```diff
@@ -107,13 +108,48 @@ def _funcRows(col, rows):  # wrap builtins so they can have a .type
 def mean(vals):
     vals = list(vals)
     if vals:
-        return float(sum(vals))/len(vals)
+        if type(vals[0]) is date:
+            vals = [d.timestamp() for d in vals]
```

Comment:
I don't think these special cases belong in the aggregators.  The idea is that people can define their own aggregators with 'obvious' implementations and they should work reasonably.  (and regardless, where these checks are, they should use `isinstance` with the Python stdlib datetime as you're doing below).

To that end, the visidata `date` and `datedelta` classes both already have `__float__` methods which convert them to float as you're doing here.  So maybe the aggregator callsite can call `aggregator.intype` on each value, and for these "more numeric" aggregators would set their intype to `float`.  Then they would need to convert the float result back into some result type, which as you've noted below can be tricky for date formats.